### DISCUSSION TITLE
VSCode F5 Run project in debug mode

### DISCUSSION THREAD

**Discussion by TonyGravagno** [araera111/mysql-to-zod#134] [Q&A] [answered]
I cannot run the project in debug mode. I have tried **many** different settings in tsconfig.json, tsconfig.build.json, package.json, and (created) launch.json. I have tried to debug the .ts files with ESM and the dist/.js files as CommonJS. Every attempt fails.

Are you running in debug mode with breakpoints and other features?

If so, **please** tell me how (maybe with a new repo branch with config file details?).

Thanks!

**Comment by araera111** [comment, marked as answer]
I’ve considered how to debug the library. While I will integrate these changes into the main branch later, here’s a temporary debugging setup that you can use:

Please place the following two files in the ./.vscode directory:

launch.json

```json:launch.json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "ts-debug",
            "program": "${workspaceFolder}/src/main.ts",
            "preLaunchTask": "prebuild",
            "outFiles": ["${workspaceFolder}/dist/**/*.js"]
        }
    ]
}
```

tasks.json

```json:tasks.json
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "npm",
            "label": "prebuild",
            "detail": "pnpm build",
            "script": "build"
        }
    ]
}
```

Please give this setup a try. I hope it works for you!

**Reply by TonyGravagno** [reply]
I see you updated the project for debugging. Thank you SO much. It's all working with a fresh pull from core so we don't need the details in this thread anymore.
